### PR TITLE
Add export scanner results autosave modal

### PR DESCRIPTION
**Pull request by carolinemodic**
Branch depends on https://github.com/votingworks/vxsuite/pull/106 as I wanted to reuse some shared components (but could be separated)

Adds a new modal when exporting scanner results that handles different USB drive states and can either save to a custom file location or save by default to `/cast-vote-records/<election name>/<scannerId><numBallots><timestamp>.jsonl`. Mimics the UI and copy of the export modal for the ballot package in election manager. When run not through kiosk mode there is an export button on the "no usb" screen so that we can still more easily save files when developing locally, it functions how the download out of kiosk mode did before this change. One small functionality change is that I am now disabling the export button when there are 0 results to export. 

I tried to break things into commits a bit more with this PR to make it a bit simpler to review. The commit messages are fairly self explanatory. Some Notes:
-   The first commit copies a pattern in the bmd code that gets the machine ID environment variable accessible from the frontend, I did this as I wanted it included in the filename. I think we should consider in the future showing this on the header/footer as well. 
- The "ui tweaks" commit has a change to autoexpand the button width in a ButtonBar when centerOnlyTrue is false (and set that in the Modal component), this is how it worked in ElectionManager but ElectionManager is on a much older version of these components code so I wasn't completely sure what was intentional.  
- The final commit is most of the actual work

## Screenshots 
(happy to make a gif as well if both are useful but I thought these would be easier for review)
### No Usb Screen
Screen Shot 2020-12-09 at 3 46 38 PM

### USB is mounted - main export screen
Screen Shot 2020-12-09 at 3 49 09 PM

### Download Complete
Screen Shot 2020-12-09 at 3 49 49 PM

### Download Complete - after eject
Screen Shot 2020-12-09 at 3 49 59 PM

### Download Failed
Screen Shot 2020-12-09 at 4 45 00 PM

### CODE REVIEW

**Comment by carolinemodic** [comment]
ccing @benadida in case you have feedback but no requirement to review :)

File: apps/bsd/src/contexts/AppContext.ts [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
 interface AppContextInterface {
   usbDriveStatus: string
   usbDriveEject: () => void
+  machineConfig: MachineConfig
 }
 
 const appContext: AppContextInterface = {
   usbDriveStatus: '',
   usbDriveEject: () => undefined,
+  machineConfig: { machineId: '000' },
```
Comment (benadida) [inline apps/bsd/src/contexts/AppContext.ts:13]:
```suggestion
  machineConfig: { machineId: '0000' },
```

File: apps/bsd/src/setupProxy.js [hunk 2]
```diff
@@ -12,4 +12,10 @@ module.exports = function (app) {
   app.use(proxy('/card', { target: 'http://localhost:3001/' }))
   app.use(proxy('/scan', { target: 'http://localhost:3002/' }))
   app.use(proxy('/config', { target: 'http://localhost:3002/' }))
+
+  app.get('/machine-config', (req, res) => {
+    res.json({
+      machineId: process.env.VX_MACHINE_ID || '000',
```
Comment (benadida) [inline apps/bsd/src/setupProxy.js:18]:
```suggestion
      machineId: process.env.VX_MACHINE_ID || '0000',
```

File: apps/bsd/src/util/filenames.ts [hunk 3]
```diff
@@ -1,11 +1,13 @@
 import moment from 'moment'
+import { Election } from '@votingworks/ballot-encoder'
 
 const SECTION_SEPERATOR = '__'
 const SUBSECTION_SEPERATOR = '_'
```
Comment (benadida) [inline apps/bsd/src/util/filenames.ts:5]: ahah, looks like this typo predates this PR :)

File: apps/bsd/test/helpers/fakeMachineConfig.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { Provider, MachineConfig } from '../../src/config/types'
+
+export default function fakeMachineConfig({
+  machineId = '000',
```
Comment (benadida) [inline apps/bsd/test/helpers/fakeMachineConfig.ts:4]:
```suggestion
  machineId = '0000',
```

**Review by benadida — APPROVED** [review]
a few questions worth considering, otherwise looks good to me.

File: apps/bsd/src/AppRoot.tsx [hunk 5]
```diff
@@ -90,6 +95,19 @@ const App: React.FC = () => {
     initialize()
   }, [refreshConfig])
 
+  useEffect(() => {
+    const initialize = async () => {
+      try {
+        const newMachineConfig = await machineConfigProvider.get()
+        setMachineConfig(newMachineConfig)
+      } catch (e) {
+        window.setTimeout(initialize, 1000)
```
[the remark this answers is not in the document]
Comment (carolinemodic) [reply apps/bsd/src/AppRoot.tsx:104]: I was copying how the useEffect block for refreshConfig works right above this, but I'm not really sure in what scenario we would see an error here, bmd just does nothing in this scenario: https://github.com/votingworks/vxsuite/blob/49ea8abe976864cda162b143c4bc39a6cf9ffc72/apps/bmd/src/AppRoot.tsx#L217

**Pull request merged**